### PR TITLE
added Nat.le_add_l

### PR DESCRIPTION
**Pull request by mrhaandi**
This PR systematically adds `le_add_l` to `NAddOrder.v` (all other lemmas in `NAddOrder.v` have both `_l` and `_r` variants).

This makes for a better drop-in replacement of the deprecated `Plus.le_plus_r`.
The deprecation message suggests to rewrite with commutativity first. This is tricky in automation and can lead to regressions.

### CODE REVIEW

**Comment by mrhaandi** [comment]
Since this makes fixing deprecation warnings smoother, I suggest including it in `8.16`.

File: theories/Arith/Arith_prebase.v
Comment (proux01) [inline theories/Arith/Arith_prebase.v:205]: Just to be safe, this should probably be deprecated instead of removed.

**Review by proux01 — APPROVED** [review]
LGTM. This is an addition in the stdlib, seems unlikely to break much but inclusion in 8.16 is up to the RM.
@ppedrot WDYT?
According to this choice, either the changelog should be moved or the deprecation version adapted before merging.

File: theories/Arith/Arith_prebase.v
Comment (mrhaandi) [reply theories/Arith/Arith_prebase.v:205]: If this PR is not included in 8.16, then I'll use the deprecation mechanism.
If this PR is included, then `le_plus_r_stt` was an undocumented ad-hoc solution to the missing lemma (which is added by this PR). So, I would like to not include `le_plus_r_stt` in 8.16 at all.

**Comment by Zimmi48** [comment]
>either the changelog should be moved

I think it is fine where it is (the changelog for 8.16.0 will have to be consolidated before the release anyway).

File: theories/Arith/Arith_prebase.v
Comment (mrhaandi) [reply theories/Arith/Arith_prebase.v:205]: Due to the argument order issue below, I again included `le_plus_r_stt` as `Nat.le_add_l` with reversed arguments.

**Comment by mrhaandi** [comment]
> So I would still add the compatibility layer `le_plus_r_stt` with the old argument order.

Done. Now the PR is very conservative, but still relevant for an 8.16 transition.

**Comment by Alizter** [comment]
cc @olaure01

**Comment by olaure01** [comment]
As this is related with #14736 it could be nice to take the opportunity to enhance the replacement of `Plus.le_plus_r` done at that time. Typically introduction of `rewrite Nat.add_comm; apply Nat.le_add_r.` in replacement of `apply le_plus_r.` in stdlib could now be `apply Nat.le_add_l.`.
As far as I can see, there are 14 calls to `le_plus_r` in stdlib 8.15.2. And they are almost all of the shape `apply le_plus_r.`.

**Comment by mrhaandi** [comment]
> As far as I can see, there are 14 calls to `le_plus_r` in stdlib 8.15.2. And they are almost all of the shape `apply le_plus_r.`.

Ok, I'll take a look.

**Review by olaure01 — APPROVED** [review]

**Comment by olaure01** [comment]
I plan to merge tomorrow.
@ppedrot @Zimmi48 I am not so familiar with post-branching merging: is it OK that I merge with milestone 8.16.0 and @ppedrot decides afterwards to backport it or not, or should I wait for a decision and adapt milestone in consequence before merging?

**Comment by olaure01** [comment]
Same PR, just possibly changelog should be adapted.

**Comment by olaure01** [comment]
@mrhaandi please squash and I'll merge

**Comment by mrhaandi** [comment]
> @mrhaandi please squash and I'll merge

Done.

**Comment by Zimmi48** [comment]
>Same PR, just possibly changelog should be adapted.

The changelog looks fine whatever the backporting decision is.

**Review by Alizter — APPROVED** [review]

**Comment by olaure01** [comment]
@coqbot merge now

**Comment by ppedrot** [comment]
For some reason backporting this PR results in a merge conflict. What happened to master in the meantime?

**Comment by mrhaandi** [comment]
It appears that https://github.com/coq/coq/commit/37f001eff51675653226d39fccd218f5eb36c962 "Strict focusing for Reals/" improves `theories/Reals/SeqSeries.v` using the previously suggested approach, which is made obsolete by this PR.
So, in 8.16 `theories/Reals/SeqSeries.v` could be left, as if not touched by this PR.

**Pull request merged**
